lib/matplotlib/backend_bases.py

```diff
@@ -2840,6 +2840,48 @@ def create_with_canvas(cls, canvas_class, figure, num):
         """
         return cls(canvas_class(figure), num)
 
+    @classmethod
+    def start_main_loop(cls):
+        """
+        Start the main event loop.
+
+        Interactive backends need to reimplement this method or
+        `~.FigureManagerBase.pyplot_show`.
```

Comment:
Can you be more specific? What is the criterion to `start_main_loop()` vs. `pyplot_show()`?

Reply:
Explained a bit more.